src/Tuura/Boolean.hs

```diff
@@ -15,9 +17,22 @@ import Tuura.Concept.Circuit
 
 newtype CNF a = CNF { fromCNF :: [[Literal a]] }
 
+instance Show (CNF String) where
+  show = eqShow "*" " + " ((flip (++) ")") . (++) "(") fromCNF
```

Comment:
It is often assumed that `show` results can be compiled. I would recommend to use derived `Show` instance here, but if you need a nicer representation for readability, you can call this function `pretty`, which seems to be a standard convention.